cmd/thanos/downsample.go

```diff
@@ -31,20 +32,50 @@ import (
 func registerDownsample(m map[string]setupFunc, app *kingpin.Application, name string) {
 	cmd := app.Command(name, "continuously downsamples blocks in an object store bucket")
 
+	httpAddr := regHTTPAddrFlag(cmd)
+
 	dataDir := cmd.Flag("data-dir", "Data directory in which to cache blocks and process downsamplings.").
 		Default("./data").String()
 
 	objStoreConfig := regCommonObjStoreFlags(cmd, "", true)
 
 	m[name] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, tracer opentracing.Tracer, _ bool) error {
-		return runDownsample(g, logger, reg, *dataDir, objStoreConfig)
+		return runDownsample(g, logger, reg, *httpAddr, *dataDir, objStoreConfig)
 	}
 }
 
+type DownsampleMetrics struct {
+	downsamples               *prometheus.CounterVec
+	downsampleFailures        *prometheus.CounterVec
+}
+
+func newDownsampleMetrics(reg *prometheus.Registry) (*DownsampleMetrics, error) {
+	m := new(DownsampleMetrics)
+
+	m.downsamples = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "thanos_compact_downsample_total",
+		Help: "Total number of downsampling attempts.",
+	}, []string{"group"})
+	m.downsampleFailures = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "thanos_compact_downsample_failures_total",
+		Help: "Total number of failed downsampling attempts.",
+	}, []string{"group"})
+
+	if err := reg.Register(m.downsamples); err != nil {
```

Comment:
This could use `MustRegister` (as done for other metrics) and skip the error check in the caller.

Reply:
Yes, `MustRegister` is ok here